packages/api-server/src/models/user.ts

```diff
@@ -0,0 +1,134 @@
+import { QueryResult } from 'pg'
+
+import { UserModel } from '@portaler/types'
+
+import { dbQuery } from '../db'
+import { DiscordMe } from '../utils/discord/fetchUser'
+import { DiscordMeGuilds } from '../utils/discord/fetchUserGuilds'
+import logger from '../utils/logger'
+
+export type UserError = 'NoUser' | 'NoServers' | 'NoRoles'
+
+/**
+ * Gets a User from the database
+ * @param  discordId
+ * @returns UserModel
+ */
+export const getUser = async (
+  discordId: string
+): Promise<UserModel | UserError> => {
+  const {
+    rows: userRows,
+  } = await dbQuery('SELECT * FROM users WHERE discord_id = $1', [discordId])
+
+  if (userRows.length === 0) {
+    logger.info('Return')
+    return Promise.resolve('NoUser')
+  }
+
+  const user: UserModel = {
+    id: userRows[0].id,
+    discordId: userRows[0].discord_id,
+    discordName: userRows[0].discord_name,
+    discordRefresh: userRows[0].discord_refresh,
+    createdOn: userRows[0].created_on,
+  }
+
+  const { rows: userServers } = await dbQuery(
+    `
+    SELECT us.server_id AS server, ur.user_role AS role
+    FROM user_servers AS us
+    JOIN user_roles AS ur ON (us.user = ur.user)
+    WHERE us.user = $1
+  `,
+    [user.id]
+  )
+
+  if (userServers.length === 0) {
+    return Promise.resolve('NoServers')
+  }
+
+  user.serverAccess = userServers.map((s) => ({
+    serverId: s.server,
+    roleId: s.role,
+  }))
```

Comment:
Wouldn't the structure here be better as just a map with server id as key and a list of roles as value? Can be created with reduce.